File: ObjcFrame-2.0/CheTuoTuo/Utils/BWUserDefaults(缓存)/BWUserDefaults.m
//
//  DWUserDefaults.m
//  Unicorn
//
//  Created by Marshall Yang on 2018/1/8.
//  Copyright © 2018年 Marshall Yang. All rights reserved.
//

#import "BWUserDefaults.h"

#define UserDefaults [NSUserDefaults standardUserDefaults]

@implementation BWUserDefaults

#pragma mark - 用户相关

/// 登录状态
+ (BOOL)userLoginState {
    if ([UserDefaults objectForKey:UserLoginState] == nil) {
        [UserDefaults setObject:@"0" forKey:UserLoginState];
        [UserDefaults synchronize];
        return NO;
    }
    NSString *state = [UserDefaults objectForKey:UserLoginState];
    if ([state isEqualToString: @"1"]) {
        return YES;
    }
    return NO;
}
+ (void)saveUserLoginState:(BOOL)state {
    if (state) {
        //登录
        [UserDefaults setObject:@"1" forKey:UserLoginState];
    } else {
        //退出登录
        [UserDefaults setObject:@"0" forKey:UserLoginState];
        //清空token
        [UserDefaults setObject:nil forKey:UserToken];
        //清空用户信息
        [UserDefaults setObject:nil forKey:Username];
        [UserDefaults setObject:nil forKey:UserNickname];
        [UserDefaults setObject:nil forKey:UserPhoneNumber];
        [UserDefaults setObject:nil forKey:UserCityName];
        [UserDefaults setObject:nil forKey:UserLocation];
    }
    [UserDefaults synchronize];
}

/// token
+ (NSString *)userToken {
    NSString *token = [UserDefaults objectForKey:UserToken];
    return BWCheckString(token);
}
+ (void)saveUserToken:(NSString *)token {
    [UserDefaults setObject:token forKey:UserToken];
    [UserDefaults synchronize];
}

/// uid
+ (NSString *)userId {
    NSString *userId = [UserDefaults objectForKey:UserID];
    return BWCheckString(userId);
}
+ (void)saveUserId:(NSString *)userId {
    [UserDefaults setObject:BWCheckString(userId) forKey:UserID];
    [UserDefaults synchronize];
}

/// 昵称
+ (NSString *)userNickName {
    NSString *nickname = [UserDefaults objectForKey:UserNickname];
    return BWCheckString(nickname);
}
+ (void)saveUserNickname:(NSString *)nickname {
    [UserDefaults setObject:BWCheckString(nickname) forKey:UserNickname];
    [UserDefaults synchronize];
}

/// 用户名
+ (NSString *)userName {
    NSString *username = [UserDefaults objectForKey:Username];
    return BWCheckString(username);
}
+ (void)saveUserName:(NSString *)userName {
    [UserDefaults setObject:BWCheckString(userName) forKey:Username];
    [UserDefaults synchronize];
}

/// 手机号
+ (NSString *)userPhoneNumber {
    NSString *phoneNumber = [UserDefaults objectForKey:UserPhoneNumber];
    return BWCheckString(phoneNumber);
}
+ (void)saveUserPhoneNumber:(NSString *)phoneNumber {
    [UserDefaults setObject:BWCheckString(phoneNumber) forKey:UserPhoneNumber];
    [UserDefaults synchronize];
}

/// 密码
+ (NSString *)userPassword {
    NSString *password = [UserDefaults objectForKey:UserPassword];
    return BWCheckString(password);
}
+ (void)saveUserPassword:(NSString *)password {
    [UserDefaults setObject:BWCheckString(password) forKey:UserPassword];
    [UserDefaults synchronize];
}

/// 用户所在城市
+ (NSString *)userCityName {
    NSString *cityName = [UserDefaults objectForKey:UserCityName];
    return BWCheckString(cityName);
}
+ (void)saveUserCityName:(NSString *)city {
    [UserDefaults setObject:BWCheckString(city) forKey:UserCityName];
    [UserDefaults synchronize];
}

/// 用户当前经纬度
+ (CLLocationCoordinate2D)userLocation {
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    NSDictionary *dict = [userDefaults objectForKey:UserLocation];
    double lat = [dict[@"lat"] doubleValue];
    double lng = [dict[@"lng"] doubleValue];
    return CLLocationCoordinate2DMake(lat, lng);
}
+ (void)saveUserLocation:(CLLocationCoordinate2D)location {
    NSDictionary *dict = @{
        @"lat" : @(location.latitude),
        @"lng" : @(location.longitude),
    };
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    [userDefaults setObject:dict forKey:UserLocation];
    [userDefaults synchronize];
}


#pragma mark - 司机相关

@end

File: SwiftFrame-1.0/MengTianXia/Others/BWUserDefaults.swift
//
//  BWUserDefaults.swift
//  MengTianXia
//
//  Created by zl on 2019/8/1.
//  Copyright © 2019 LuMeng. All rights reserved.
//

import UIKit

// 用户登录状态
let CustomerState = "CUSTOMERSTATE"
let UserToken = "USERTOKEN"
let UserId = "USERID"
let PhoneNumber = "PHONENUMBER"
let UserName = "USERNAME"


class BWUserDefaults: NSObject {
    // 单例
    static let shareInstance = BWUserDefaults()
    
    /// 获取用户登录状态
    public func getCustomerState() -> Bool {
        let state = UserDefaults.standard.object(forKey: CustomerState) as? String
        if state == nil {
            UserDefaults.standard.set("0", forKey: CustomerState)
            UserDefaults.standard.synchronize()
            return false
        }
        if state == "1" {
            return true
        }
        return false
    }

    /// 保存用户登录状态
    public func setCustomerState(state: Bool) {
        if state {
            // 登录
            UserDefaults.standard.set("1", forKey: CustomerState)
        } else {
            // 退出登录
            UserDefaults.standard.set("0", forKey: CustomerState)
            // 清除数据
            clearCustomerData()
        }
    }
    
    
    /// uid
    public func getUserId() -> String? {
        let userId = UserDefaults.standard.object(forKey: UserId) as? String
        return userId
    }
    public func setUserId(userId: String) {
        UserDefaults.standard.set(userId, forKey: UserId)
        UserDefaults.standard.synchronize()
    }
    
    /// token
    public func getUserToken() -> String? {
        let token = UserDefaults.standard.object(forKey: UserToken) as? String
        return token
    }
    public func setUserToken(token: String) {
        UserDefaults.standard.set(token, forKey: UserToken)
        UserDefaults.standard.synchronize()
    }
    
    /// 手机号
    public func getPhoneNumber() -> String? {
        let phone = UserDefaults.standard.object(forKey: PhoneNumber) as? String
        return phone
    }
    public func setPhoneNumber(number: String) {
        UserDefaults.standard.set(number, forKey: PhoneNumber)
        UserDefaults.standard.synchronize()
    }
    
    /// 用户名
    public func getUserName() -> String? {
        let name = UserDefaults.standard.object(forKey: UserName) as? String
        return name
    }
    public func setUserName(name: String) {
        UserDefaults.standard.set(name, forKey: UserName)
        UserDefaults.standard.synchronize()
    }
}


// MARK: - 私有方法
extension BWUserDefaults {
    /// 清除用户数据
    func clearCustomerData() {
        UserDefaults.standard.set("0", forKey: CustomerState)
        UserDefaults.standard.set("", forKey: UserToken)
        UserDefaults.standard.set("", forKey: UserId)
        UserDefaults.standard.set("", forKey: PhoneNumber)
        UserDefaults.standard.set("", forKey: UserName)
    }
}
